### PR TITLE
[Charity-59] add endpoints to work with charity organisation managers

### LINKED ISSUE #59: [Charity] Adding managers to organisation
**Description** 
As a user, I want to delegate charity organisation managing to other people. Manager can be usual and with extended roots. Also, I want to see all managers in exact company. 

**Acceptance criteria** 
- [x] `/charities/{organisation_id}/managers/add` should add manager to excising organisation
- [x] `/charities/{organisation_id}/managers` should retrieve all managers of current organisation
- [x] `/charities/{organisation_id}/managers/{user_id}` should delete manager from the company
- [x] code should be covered by tests

### PR DESCRIPTION
**Pull request by paivazov**
This pull request implements endpoints with possibility to add, delete, show current managers of charity organisation and small refactoring.

**References**
- #59

### CODE REVIEW

File: api_server/charity/routers.py [hunk 1]
```diff
@@ -101,9 +104,54 @@ async def delete_charity(org_id: UUID, charity_service: CharityService = Depends
...
-    ResponseBaseSchema object with CharityOutputSchema object as response data.
+        ResponseBaseSchema object with CharityOutputSchema object as response data.
     """
     await charity_service.delete_organisation(org_id)
     return ResponseBaseSchema(status_code=HTTP_200_OK,
                               data={"detail": "Organisation has been deleted successfully."},
                               errors=[])
+
+
+@charities_router.post("/{organisation_id}/managers/add", response_model=ResponseBaseSchema)
```
Comment (BorodaUA) [inline api_server/charity/routers.py]: That ```/add``` at the end is absolutely unnecessary and not restful.
Some good read: https://restfulapi.net/resource-naming/

File: api_server/charity/services.py [hunk 2]
```diff
@@ -89,50 +91,95 @@ async def delete_organisation(self, org_id: UUID):
         """
         return await self._delete_organisation(org_id)
 
-    @classmethod
-    async def _get_organisation_by_id(cls, org_id: UUID) -> CharityOrganisation:
+    async def add_manager(self, organisation_id: UUID, manager: AddManagerSchema) -> CharityUserAssociation:
+        """
+        Delete CharityOrganisation object.
```
Comment (BorodaUA) [inline api_server/charity/services.py]: Something with docstring I guess.

File: api_server/charity/services.py [hunk 3]
```diff
@@ -89,50 +91,95 @@ async def delete_organisation(self, org_id: UUID):
...
+            organisation_id: id of organisation we want to list of managers.
+
+        Returns:
+             InstrumentedList object.
+        """
+        return await self._get_managers_of_organisation(organisation_id)
+
+    async def delete_manager_from_organisation(self, organisation_id: UUID, user_id: UUID):
+        """
+        Retrieves list of users that related to definite organisations.
```
Comment (BorodaUA) [inline api_server/charity/services.py]: Something with docstring I guess.

File: api_server/charity/tests/test_charity.py [hunk 4]
```diff
@@ -89,11 +92,12 @@ async def test_create_charity_should_be_ok(
...
-                                                                   ):
+    async def test_edit_charity_not_authenticated_should_be_denied(
+            self,
+            app: FastAPI,
+            client: AsyncClient,
+            charity: CharityOrganisation
+    ):
         """
         Test PATCH '/charities/{org_id}' endpoint with NOT authenticated user.
         """
```
Comment (BorodaUA) [inline api_server/charity/tests/test_charity.py]: Looks like incompleate docstring.

File: api_server/charity/tests/test_charity.py [hunk 5]
```diff
@@ -176,12 +184,30 @@ async def test_edit_charity_by_random_user_should_be_forbidden(self,
...
-                                                                     charity: CharityOrganisation,
-                                                                     authenticated_random_test_user: User
-                                                                     ):
+    async def test_edit_charity_with_duplicate_title_should_be_forbidden(
+            self,
+            app: FastAPI,
+            client: AsyncClient,
+            many_charities: List[CharityOrganisation],
+    ):
+        charity = None
```
Comment (BorodaUA) [inline api_server/charity/tests/test_charity.py:203]: No docstring at all?

File: api_server/charity/tests/test_charity.py [hunk 6]
```diff
@@ -299,3 +342,127 @@ async def test_show_existed_charities_should_be_ok(self,
...
+            charity_with_authenticated_manager: CharityOrganisation,
+            random_test_user: User):
+        """
+        Test POST '/charities/{org_id}/managers/add' endpoint. Should be denied because manager has not
+        supermanager rights.
+        Args:
+                   app: pytest fixture, an instance of FastAPI.
+                   client: pytest fixture, an instance of AsyncClient for http requests.
+                   charity_with_authenticated_manager: pytest fixture,
+                   add CharityOrganisation to database with non-authenticated user
```
Comment (BorodaUA) [inline api_server/charity/tests/test_charity.py:394]: Looks like no "random_test_user" fixture here.

**Review by BorodaUA — CHANGES_REQUESTED** [review]

File: api_server/charity/tests/test_charity.py (hunk 4 above)
Comment (paivazov) [reply api_server/charity/tests/test_charity.py]: Okay, I will make it more descriptive

File: api_server/charity/tests/test_charity.py (hunk 5 above)
Comment (paivazov) [reply api_server/charity/tests/test_charity.py:203]: I've added docstring

**Review by BorodaUA — APPROVED** [review]

**Review by evroandrew — APPROVED** [review]

**Review by MaksymShvets — APPROVED** [review]

**Pull request merged**